sensor/common/sensor/central_receiver_impl.go

```diff
@@ -99,8 +99,7 @@ func (s *centralReceiverImpl) receive(stream central.SensorService_CommunicateCl
 
 func sendToAll(ctx context.Context, msg *central.MsgToSensor, wg *sync.WaitGroup, componentsQueues map[string]chan *central.MsgToSensor) {
 	wg.Add(len(componentsQueues))
-	ctx, cancel := context.WithTimeout(ctx, time.Second)
-	defer cancel()
+	ctx, _ = context.WithTimeout(ctx, time.Second)
```

Comment:
Hey, I saw your comment on Slack saying that this solves an issue with Integration test. 
I am wondering why you decided on this version instead of waiting for the `wg` to finish all jobs and cancel the context afterwards?

Reply:
so you propose to use something like:
```
defer func() {
    wg.Wait()
    cancel()
}
```
I'm not sure if this make sense here as context is local to this function and used to not send messages after parent cancel context and drop messages when component queue is full

Reply:
I proposed an alternative in a separate comment. PTAL.